frontend/src/components/App/PluginSettings/PluginSettings.tsx

```diff
@@ -284,15 +275,60 @@ export function PluginSettingsPure(props: PluginSettingsPureProps) {
 export default function PluginSettings() {
   const dispatch = useDispatch();
 
-  const pluginSettings = useTypedSelector(state => state.plugins.pluginSettings);
+  const pluginData = useTypedSelector(state => state.plugins.pluginData);
+
+  const oldLocalEnabledList = localStorage.getItem('headlampPluginSettings');
+  const localEnabledList = localStorage.getItem('enabledPluginsList');
+
+  let pluginsEnabledMap;
+
+  /**
+   * Note: this is a temporary fix to handle the migration of the old plugin settings.
+   */
+  if (oldLocalEnabledList) {
+    const oldSettings = JSON.parse(oldLocalEnabledList);
+
+    pluginsEnabledMap = oldSettings.reduce(
+      (acc: Record<string, any>, p: { name: string; isEnabled: boolean }) => {
+        acc[p.name] = !!p.isEnabled;
+        return acc;
+      },
+      {} as Record<string, any>
+    );
+
+    localStorage.setItem('enabledPluginsList', JSON.stringify(pluginsEnabledMap));
+    localStorage.removeItem('headlampPluginSettings');
+
+    dispatch(setEnablePlugin(pluginsEnabledMap));
+    dispatch(reloadPage());
+  } else {
+    /**
+     * NOTE: For compatibility with old settings, we need to check if the `localEnabledList` exists.
```

Comment:
So we have 2 deprecated items? headlampPluginSettings + enabledPluginsList?
I thought this was just about fixing the former, where we were storing the package.json contents for no reason, when what we were apparently using that list for was just to check if the plugins were enabled.